core/src/main/java/io/parsingdata/metal/expression/comparison/ComparisonExpression.java

```diff
@@ -57,7 +57,7 @@ public ComparisonExpression(final ValueExpression value, final ValueExpression p
 
     @Override
     public boolean eval(final ParseGraph graph, final Encoding encoding) {
-        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(Optional.of(graph.current())) : value.eval(graph, encoding);
+        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(graph.current().map(parseValue -> parseValue)) : value.eval(graph, encoding);
```

Comment:
@rdvdijk I was actually hoping something like `Function.identity()` would exist but couldn't find it late last night :) I agree that would be much better.

@ccreeten I agree that could make the code nicer by removing the need for such a conversion. But at the same time it's also good to have easy-to-understand signatures.